### PR TITLE
Notice: Add appropriate size props to Buttons

### PR DESCRIPTION
**Pull request by mirka**
In preparation for #65751

## What?

Adds appropriate size props to the remaining Buttons in the Notice component.

## Testing Instructions

See the Notice stories in Storybook.

### CODE REVIEW

File: packages/components/src/notice/index.tsx [hunk 1]
```diff
@@ -160,6 +161,7 @@ function Notice( {
 			</div>
 			{ isDismissible && (
 				<Button
+					size="small"
```
Comment (mirka) [inline packages/components/src/notice/index.tsx:164]: The Notice will naturally reduce to a shorter height, due to the smaller button. This matches the current [Figma specs](https://www.figma.com/design/804HN2REV2iap2ytjRQ055/Core-System-Library?node-id=15877-6194&t=Km1k2S0n23fJrjrO-4), so I'm going to assume this is correct (cc @WordPress/gutenberg-design).

| Before | After |
|--------|-------|
|Notice close button, before|Notice close button, after|

File: packages/components/src/notice/index.tsx [hunk 2]
```diff
@@ -142,6 +142,7 @@ function Notice( {
 
 							return (
 								<Button
+									__next40pxDefaultSize
```
Comment (mirka) [inline packages/components/src/notice/index.tsx:145]: | Before | After |
|--------|-------|
|Notice action buttons, before|Notice action buttons, after|

File: packages/components/src/notice/stories/index.story.tsx [hunk 3]
```diff
@@ -109,7 +109,11 @@ export const NoticeListSubcomponent: StoryFn< typeof NoticeList > = () => {
 	return (
 		<>
 			<NoticeList notices={ notices } onRemove={ removeNotice } />
-			<Button variant="primary" onClick={ resetNotices }>
+			<Button
+				__next40pxDefaultSize
```
Comment (mirka) [inline packages/components/src/notice/stories/index.story.tsx:113]: Reset button in story

Comment (tyxla) [reply packages/components/src/notice/stories/index.story.tsx:113]: Pre-existing, but isn't it weird that those notices have no margins by default?

File: packages/components/src/notice/index.tsx (hunk 1 above)
Comment (tyxla) [reply packages/components/src/notice/index.tsx:164]: I'm good with this, but I'd love to get a design confirmation before we 🚢 it.

File: packages/components/src/notice/index.tsx (hunk 2 above)
Comment (tyxla) [reply packages/components/src/notice/index.tsx:145]: LGTM in the template validation notice:

Screenshot 2024-10-30 at 14 49 41

**Review by tyxla — APPROVED** [review]
LGTM from my perspective, tests well with notices I've tested across the editor 👍 🚀

File: packages/components/src/notice/index.tsx (hunk 1 above)
Comment (jameskoster) [reply packages/components/src/notice/index.tsx:164]: Seems fine until we holistically revisit the Notice design.

File: packages/components/src/notice/stories/index.story.tsx (hunk 3 above)
Comment (mirka) [reply packages/components/src/notice/stories/index.story.tsx:113]: Not sure 🤷 This component is on the "needs design updates" docket though.

**Pull request merged**